File: src/delete_weather.py
import json
import os

import boto3

dynamodb = boto3.resource('dynamodb')
table_name = os.environ.get("TABLE_NAME")


def lambda_handler(event, context):
    table = dynamodb.Table(table_name)
    key = json.loads(event['pathParameters'])['id']
    id_key = {
        "id": key
    }
    print(event)
    try:
        table.delete_item(Key=id_key)
        return {
            'statusCode': 200,
            'body': "Weather Delete Succeessfull"
        }
    except table_name:
        return {
            'statusCode': 500,
            'body': "An error occured. Weather not deleted"
        }

File: packages/functions/src/deleteWeather.ts
import { DynamoDB } from "aws-sdk";
import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";

const dynamoDb = new DynamoDB.DocumentClient();
const tableName = process.env.TABLENAME
export async function main(
  event: APIGatewayProxyEvent
): Promise<APIGatewayProxyResult> {
  const weatherId =
    event.pathParameters && event.pathParameters.id
      ? event.pathParameters.id
      : null;

  if (!weatherId) {
    return {
      statusCode: 404,
      body: JSON.stringify({ error: true }),
    };
  }else{
    dynamoDb.delete({TableName:tableName, Key:{
        id: weatherId
    }}).promise()
    return {
      statusCode: 200,
      body: JSON.stringify({
        "message": "Weather deleted successfully"
      }),
    };
  }
}